ambari-server/src/main/java/org/apache/ambari/server/controller/internal/ViewURLResourceProvider.java

```diff
@@ -275,6 +274,11 @@ public Void invoke() throws AmbariException {
           throw new AmbariException("This view URL name exists, URL names should be unique");
         }
 
+        savedUrl = viewURLDAO.findBySuffix(urlEntity.getUrlSuffix());
```

Comment:
This can be written as 

```java
savedUrl = viewURLDAO.findBySuffix(urlEntity.getUrlSuffix()).orElseThrow(() -> new AmbariException("This view URL suffix has already been recorded, URL suffixes should be unique"));
```

Reply:
That would be true if we used Java's own `Optional` class here. However we import Guava's class in this class. In addition to this I also believe that my approach is good enough and easy to read and understand to everyone. I'll make one change: we do not need the local variable called 'savedUrl' since it's not used going forward so that we can inline that DAO method call within the `if` statement.